index.js

```diff
@@ -1,267 +1,260 @@
-var fs = require('fs');
-var when = require('when');
-var util = require('util');
-var Port = require('ut-bus/port');
-var errors = require('./errors');
-var FtpClient;
+'use strict';
+const util = require('util');
+const fs = require('fs');
+let FtpClient;
 
-function FtpPort() {
-    Port.call(this);
-    this.config = {
-        id: null,
-        logLevel: '',
-        type: 'ftp'
-    };
-    /**
-     * @param {Object} client
-     * @description holds the FTP client instance
-     */
-    this.client = null;
-    this.ready = false;
-    this.reconnectInterval = null;
-    return this;
-}
-
-util.inherits(FtpPort, Port);
-
-/**
- * @function init
- * @description Extends the default Port.init() method and determines which ftp client to require
- */
-FtpPort.prototype.init = function init() {
-    Port.prototype.init.apply(this, arguments);
-    this.latency = this.counter && this.counter('average', 'lt', 'Latency');
-
-    if (this.config.protocol === 'sftp') {
-        FtpClient = require('scp2/lib/client').Client;
-    } else {
-        FtpClient = require('ftp');
+module.exports = function({parent}) {
+    function FtpPort({config}) {
+        parent && parent.apply(this, arguments);
+        this.config = Object.assign({
```

Comment:
We have to use lodash.merge when the object we merge into is not flat. 
For example doing
```js
Object.assign({host:'localhost',port:22}, {port: 1000})
```
will overwrite the default host with undefined host.